### PR TITLE
Api guidelines

### PR DESCRIPTION
**Pull request by MarcosX**
This PR includes conventions for Pagination and Versioning.

Since this is the first PR we are specially looking for feedback regarding the general format and directory structure.

#1

### CODE REVIEW

File: api/pagination.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    },
+    {
+      "id": 2,
+      ...
+    },
+    ...
+  ]
+ "links": {
+    "prev": "http://localhost:3000/leave?page=1&per_page=1",
+    "next": "http://localhost:3000/leave?page=1&per_page=3"
```
Comment (MarcosX) [inline api/pagination.md]: I think you changed the wrong parameter in the example (`per_page` instead of `page`).

Comment (lucmir) [inline api/pagination.md]: Corrected. Thanks!

**Comment by Stono** [comment]
Hey @MarcosX and @lucmir - great work getting this merged.
Maybe we could get into the habit however of seeking feedback from other people outside of the staffing team before accepting the merge requests?

Looking at the stuff you've written, I certainly have some comments which I would have liked to explore as part of the merge & review.

Maybe even showcase on the tech huddle to gather feedback?

**Comment by MarcosX** [comment]
hey @Stono, we posted it three times in the #techops-platform chanel and @lucmir brought it up during standup.

Maybe we didn't do a good enough job of communicating that we wanted feedback or maybe our tools are not serving as well. Anyways we'll try to improve it going forward, but this PR was open since yesterday and we felt that if we didn't merge it we would end up with a huge PR that no one would review.

Feel free to make your changes, the branch is merged but this is a live document that will always be incomplete :)

**Comment by Stono** [comment]
Hey, I wasn't having a go! It was just a suggestion, don't be so defensive
:-)  I think the request got lost across multiple channels and that's bound
to happen whilst we find our best ways of working.

In GitHub if you're seeking review from others you can add multiple people
to the "reviewers" section.   This then sends them a GitHub notification.
This is quite a good way to "prompt" people.

Ta

**Comment by MarcosX** [comment]
Cool! Learned two things: didn't know the "having a go" expression and didn't know about adding reviewers to a PR.

Sorry about being defensive, english is my second language and it's hard to express myself politely sometimes (and thanks for the feedback!).

**Comment by Stono** [comment]
Haha no problem.  Its actually good feedback for me too.  We all need to be
conscious of how things can be read!

Speak to you tomorrow :)

**Pull request merged**